### PR TITLE
Adding ChainFileListFilter constructors to behave like CompositeFileL…

### PR DESCRIPTION
**Pull request by lrk**
…istFilter

Fixing https://github.com/spring-projects/spring-integration/issues/2569
Adding constructors in ChainFileListFilter matching super, so this class behave like CompositeFileListFilter when using XML flow configuration.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@lrk Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2572)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2572) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@lrk Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2572)!

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java [hunk 1]
```diff
@@ -31,11 +32,25 @@
  *
  * @author Artem Bilan
  * @author Gary Russell
+ * @author Cengis KOCAURLU
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: Do you really always write your last name with only capital letters?
Can't it be used just formal as all others?

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java [hunk 2]
```diff
@@ -31,11 +32,25 @@
...
  *
  * @since 4.3.7
  *
  */
 public class ChainFileListFilter<F> extends CompositeFileListFilter<F> {
+	/**
+	* {@inheritDoc}
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: I think we can leave without these Javadocs or let's add some formal message as "The constructor matching the super class."

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java [hunk 3]
```diff
@@ -31,11 +32,25 @@
...
+	*/
+	public ChainFileListFilter() {
+		super();
+	}
+
+	/**
+	* {@inheritDoc}
+	*/
+	public ChainFileListFilter(Collection<? extends FileListFilter<F>> fileFilters) {
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: May we have some simple test-case with the mentioned in the issue XML config?

**Review by artembilan — CHANGES_REQUESTED** [review]
Even if the fix is too simple, no limits for improvements.

Please, consider my comments or let's us know your vision on the matter.

Anyway: thank you for the contribution so far!

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java (hunk 1 above)
Comment (lrk) [reply spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: Sorry, it's just an old habit

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java (hunk 2 above)
Comment (lrk) [reply spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: i removed this unnecessary javadoc

File: spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java (hunk 3 above)
Comment (lrk) [reply spring-integration-file/src/main/java/org/springframework/integration/file/filters/ChainFileListFilter.java]: i updated the `ChainFileListFilterIntegrationTests`file

**Comment by artembilan** [comment]
@lrk ,

Thank you very much for the contribution.
Looking forward for more!

**Pull request merged**